src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/TelemetryClientService.kt

```diff
@@ -153,6 +154,23 @@ class TelemetryClientService(
     return flagEventDataMap
   }
 
+  fun createUnFlagEventForVisit(
+    visitReference: String,
+    notificationEventType: NotificationEventType?,
+    reason: UnFlagEventReason,
+  ): MutableMap<String, String> {
+    val unFlagEventDataMap = mutableMapOf(
+      "reference" to visitReference,
+      "reason" to reason.desc,
+    )
+
+    notificationEventType?.let {
+      unFlagEventDataMap["reviewType"] = it.reviewType
```

Comment:
in some instances only 1 specific event might be unflagged. e.g. exclude date removed.